pavelib/tests.py

```diff
@@ -196,44 +196,21 @@ def coverage(options):
     """
     compare_branch = getattr(options, 'compare_branch', 'origin/master')
 
-    for directory in Env.LIB_TEST_DIRS + ['cms', 'lms']:
-        report_dir = Env.REPORT_DIR / directory
-
-        if (report_dir / '.coverage').isfile():
-            # Generate the coverage.py HTML report
-            sh("coverage html --rcfile={dir}/.coveragerc".format(dir=directory))
-
-            # Generate the coverage.py XML report
-            sh("coverage xml -o {report_dir}/coverage.xml --rcfile={dir}/.coveragerc".format(
-                report_dir=report_dir,
-                dir=directory
-            ))
-
+    report_dir = Env.REPORT_DIR
+    rcfile = Env.PYTHON_COVERAGERC
+
+    if not (report_dir / '.coverage').isfile():
+        # This may be that the coverage files were generated using -p,
+        # try to combine them to the one file that we need.
+        sh("coverage combine --rcfile={}".format(rcfile))
```

Comment:
Since if there are no coverage files to combine, an empty .coverage file will be created, add a check for the filesize of the .coverage file. If it is zero bytes, fail the task intentionally with a helpful message rather than letting the next command fail.